### PR TITLE
[SPARK-8647][MLlib] Potential issue with constant hashCode

### PR DESCRIPTION
**Pull request by aloknsingh**
I added the code, 
  // see [SPARK-8647], this achieves the needed constant hash code without constant no.
  override def hashCode(): Int = this.getClass.getName.hashCode()

does getting the constant hash code as per jira

### CODE REVIEW

**Comment by srowen** [comment]
At best, this exchanges one number for another, so does that help anything? This changes semantics a bit if there are any subclasses. I don't think this is something to change without a clear motivation. At best a comment is warranted.

**Comment by MechCoder** [comment]
Could you please explain what the issue is? Every instance of `MatrixUDT` should have the same hash code and it should not be a problem whatever this is.

**Comment by mengxr** [comment]
I think @aloknsingh 's points are valid. Both `1994` and `7919` are magic numbers, which we should avoid in the code.

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala [hunk 1]
```diff
@@ -193,7 +193,8 @@ private[spark] class MatrixUDT extends UserDefinedType[Matrix] {
     }
   }
 
-  override def hashCode(): Int = 1994
+  // see [SPARK-8647], this achieves the needed constant hash code without constant no.
+  override def hashCode(): Int = this.getClass.getName.hashCode()
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala]: Would `this.getClass.hashCode` be sufficient?

**Comment by mengxr** [comment]
ok to test

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala (hunk 1 above)
Comment (aloknsingh) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala]: @mengxr agree I will change as per comment

Comment (aloknsingh) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala]: @srowen @MechCoder Please see the jira https://issues.apache.org/jira/browse/SPARK-8647  for more details.

**Comment by srowen** [comment]
LGTM in the sense that it just changes one number for another.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**